src/visitors/neuron_solve_visitor.hpp

```diff
@@ -59,6 +59,8 @@ class NeuronSolveVisitor: public AstVisitor {
     /// the derivative name currently being visited
     std::string derivative_block_name;
 
+    std::vector<std::shared_ptr<ast::Statement>> euler_values_computation;
```

Comment:
Just a nitpick.. This variable could be renamed to `euler_solution_expressions`?